### PR TITLE
Change sort logic for autofill credentials

### PR DESCRIPTION
**Pull request by CDRussell**
Task/Issue URL: https://app.asana.com/0/488551667048375/1203892213154874/f

### Description
Updates the sorting logic for autofill login credentials. When considering what to sort on for a given login credential

- title is  preferred way to sort
- if a title isn't available, or two titles are identical, then domain is considered
- if two domains are identical, then subdomain is also considered

### Steps to test this PR
- [x] Manually add the login entries detailed in https://app.asana.com/0/488551667048375/1203892213154874/f
- [x] In the credential management screen, verify credentials are sorted in the same order as that task
- [x] Add any other combinations of titles, domains and subdomains to test the sort order matches expectations

### CODE REVIEW

**Comment by CDRussell** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #2817** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/2817" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/2817?utm_source=stack-comment).

File: autofill/autofill-api/src/main/java/com/duckduckgo/autofill/api/urlmatcher/AutofillUrlMatcher.kt [hunk 1]
```diff
@@ -14,11 +14,14 @@
  * limitations under the License.
  */
 
-package com.duckduckgo.autofill.api.ui.urlmatcher
+package com.duckduckgo.autofill.api.urlmatcher
```
Comment (CDRussell) [inline autofill/autofill-api/src/main/java/com/duckduckgo/autofill/api/urlmatcher/AutofillUrlMatcher.kt:17]: part of deleting the duplicate `AutofillUrlMatcher`

File: autofill/autofill-api/src/main/java/com/duckduckgo/autofill/api/urlmatcher/AutofillUrlMatcher.kt [hunk 2]
```diff
@@ -14,11 +14,14 @@
...
 
 interface AutofillUrlMatcher {
-    fun extractUrlPartsForAutofill(originalUrl: String): ExtractedUrlParts
-    fun matchingForAutofill(visitedSite: ExtractedUrlParts, savedSite: ExtractedUrlParts): Boolean
+    fun extractUrlPartsForAutofill(originalUrl: String?): ExtractedUrlParts
```
Comment (marcosholgado) [inline autofill/autofill-api/src/main/java/com/duckduckgo/autofill/api/urlmatcher/AutofillUrlMatcher.kt:20]: nit: appreciate these methods were not documented before but would be nice to document them :)

**Review by marcosholgado — APPROVED** [review]
LGTM! Left a nit but feel free to ignore.

File: autofill/autofill-api/src/main/java/com/duckduckgo/autofill/api/urlmatcher/AutofillUrlMatcher.kt (hunk 2 above)
Comment (CDRussell) [reply autofill/autofill-api/src/main/java/com/duckduckgo/autofill/api/urlmatcher/AutofillUrlMatcher.kt:20]: Good suggestion. Going to pull this class apart soon as it's doing too much, so will be sure to document the output of that.

**Pull request merged**